src/evolveP.js

```diff
@@ -23,9 +26,10 @@ const evolveP = (transforms, obj) => {
       .then(pair(key))
   }
 
-  return Promise.resolve(toPairs(obj))
-    .then(mapP(transform))
-    .then(fromPairs)
+  return compose(
+    composeP(fromPairs, mapP(transform), resolve),
+    toPairs
+  )
```

Comment:
Changing reading direction mid-flight here is a little confusing.  I'd prefer to either have vertical or horizontal:
```js
return pipe(
  toPairs,
  pipeP(
    mapP(transform),
    fromPairs
  )
)
```
or
```js
return compose(composeP(fromPairs, mapP(transform)), toPairs)
```